health/server_internal_test.go

```diff
@@ -81,3 +86,65 @@ func (s) TestShutdown(t *testing.T) {
 		t.Fatalf("status for %s is %v, want %v", testService, status, healthpb.HealthCheckResponse_NOT_SERVING)
 	}
 }
+
+// TestList verifies that List() returns the health status of all the services if no. of services are within
+// maxAllowedLimits
+func (s) TestList(t *testing.T) {
+	s := NewServer()
+	s.mu.Lock()
+	// Remove the zero value
+	delete(s.statusMap, "")
+	// Fill out status map with information
+	for i := 1; i <= 3; i++ {
+		s.statusMap[fmt.Sprintf("%d", i)] = healthpb.HealthCheckResponse_SERVING
+	}
+	s.mu.Unlock()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	var in healthpb.HealthListRequest
+	out, err := s.List(ctx, &in)
+
+	if err != nil {
+		t.Fatalf("s.List(ctx, &in) returned err %v, want nil", err)
+	}
+	if len(out.GetStatuses()) != len(s.statusMap) {
+		t.Fatalf("len(out.GetStatuses()) = %d, want %d", len(out.GetStatuses()), len(s.statusMap))
+	}
+	for key := range out.GetStatuses() {
+		v, ok := s.statusMap[key]
+		if !ok {
+			t.Fatalf("key %s does not exist in s.statusMap", key)
+		}
+		if v != healthpb.HealthCheckResponse_SERVING {
+			t.Fatalf("%s returned status %d, want %d", key, healthpb.HealthCheckResponse_SERVING, v)
+		}
+	}
```

Comment:
Please replace this with something that confirms it is what we expect it to be statically, not what exists in `s.statusMap` dynamically.  I.e.

```go
want := healthpb.HealthListRequest{
	Statuses: map[string]*healthpb.HealthCheckResponse{
		"1": healthpb.HealthCheckResponse_SERVING.Enum(), // etc
	}
}
if diff := cmp.Diff(got, want); diff != "" {
	t.Fatalf("....: %s", diff)
}
```